tests/functional/modules/test_zos_archive_func.py

```diff
@@ -0,0 +1,935 @@
+                                        format=dict(
+                                            name=format
+                                        ))
+        
+        for result in archive_result.contacted.values():
+            print(result)
+            assert result.get("failed", False) is True
+            assert result.get("changed") is False
+        
+        set_uss_test_env(hosts, USS_TEST_FILES)
+        archive_result = hosts.all.zos_archive( path=list(USS_TEST_FILES.keys()),
+                                        dest=dest,
+                                        format=dict(
+                                            name=format
+                                        ),
+                                        force=True,)
+
+        for result in archive_result.contacted.values():
+            print(result)
+            assert result.get("failed", False) is False
+            assert result.get("changed") is True
+            assert result.get("dest_state") == expected_state
+
+
+    finally:
+        hosts.all.file(path=f"{USS_TEMP_DIR}", state="absent")
+
+
+@pytest.mark.uss
+@pytest.mark.parametrize("format", USS_FORMATS)
+@pytest.mark.parametrize("path", [
+    dict(files= f"{USS_TEMP_DIR}/*.txt", size=len(USS_TEST_FILES)),
+    dict(files=list(USS_TEST_FILES.keys()),  size=len(USS_TEST_FILES)), 
+    dict(files= f"{USS_TEMP_DIR}/" , size=len(USS_TEST_FILES) + 1), 
+    ])
+def test_uss_archive_multiple_files(ansible_zos_module, format, path):
+    try:
+        hosts = ansible_zos_module
+        hosts.all.file(path=USS_TEMP_DIR, state="absent")
+        hosts.all.file(path=USS_TEMP_DIR, state="directory")
+        set_uss_test_env(hosts, USS_TEST_FILES)
+        dest = f"{USS_TEMP_DIR}/archive.{format}"
+        expected_state = STATE_ARCHIVED
+        archive_result = hosts.all.zos_archive( path=path.get("files"),
+                                        dest=dest,
+                                        format=dict(name=format),)
+
+        # resulting archived tag varies in size when a folder is archived using zip.
+        size = path.get("size")
+        
+        for result in archive_result.contacted.values():
+            print(result)
+            assert result.get("changed") is True
+            assert result.get("dest_state") == expected_state
+            assert len(result.get("archived")) == size
+            # Command to assert the file is in place
+            cmd_result = hosts.all.shell(cmd=f"ls {USS_TEMP_DIR}")
+            for c_result in cmd_result.contacted.values():
+                assert f"archive.{format}" in c_result.get("stdout")
+                
+    finally:
+        hosts.all.file(path=USS_TEMP_DIR, state="absent")
+
+
+@pytest.mark.uss
+@pytest.mark.parametrize("format", USS_FORMATS)
+@pytest.mark.parametrize("path", [
+    dict(files=list(USS_TEST_FILES.keys()),  size=len(USS_TEST_FILES) - 1, exclude_path=[f'{USS_TEMP_DIR}/foo.txt']), 
```

Comment:
Only one option of parameters. My suggestion would be declare before the try these variables, besides are global ones.
Also delete print statements in 238, 275 and 304.

Reply:
Agree with your comment about the parameters, but if @fernandofloresg thinks this test will probably include other combinations at a later time, I think it's fine to leave it as it is.